src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -438,13 +438,14 @@ public void analyse(Network network, List<PropagatedContingency> contingencies,
                         : Collections.emptyList();
 
                 // run DC load on pre-contingency network
-                DenseMatrix flowStates = calculateFlowStates(loadFlowContext, participatingElements, new DisabledNetwork(), reportNode);
-
+                DenseMatrix baseFlowStates = calculateFlowStates(loadFlowContext, participatingElements, new DisabledNetwork(), reportNode);
+                DenseMatrix flowStates = (DenseMatrix) baseFlowStates.copy(new DenseMatrixFactory());
                 // compute the pre-contingency factor states
```

Comment:
It seems useless to copu the values here since they will be restored to the base value at each iteration. You could just create it and avoid here the iteration on all non null values (can be important on large matrixes).

Reply:
It would also be clearer to rename the variable flowStatesWorkingMatrix to make it clear that is is a reused matrix that is modified at each iteration.